### PR TITLE
Feature/class edits

### PR DESCRIPTION
**Pull request by arminWel**
This branch implements the feature, that class edits may be synchronized.

### CODE REVIEW

File: packages/CollabCodeEdit-Client.package/CCEClassDefinition.class/class/superclass.subclass.instVarNames.classVarNames.poolDictionaries.category.comment..st [hunk 1]
```diff
@@ -0,0 +1,12 @@
+as yet unclassified
```
Comment (arminWel) [inline packages/CollabCodeEdit-Client.package/CCEClassDefinition.class/class/superclass.subclass.instVarNames.classVarNames.poolDictionaries.category.comment..st]: .

File: packages/CollabCodeEdit-Client.package/CCEClassDefinition.class/class/superclass.subclass.instVarNames.classVarNames.poolDictionaries.category.comment..st [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+	^ self new
+		superclass: aClass;
+		subclass: aSymbol;
+		instVarNames: instVarString;
+		classVarNames: classVarString;
+		poolDictionaries: poolDictString;
+		category: categoryString;
+		comment: commentString
+		
+
```
Comment (arminWel) [inline packages/CollabCodeEdit-Client.package/CCEClassDefinition.class/class/superclass.subclass.instVarNames.classVarNames.poolDictionaries.category.comment..st]: .

File: packages/CollabCodeEdit-Client.package/CCEClassDefinition.class/instance/^equals.st [hunk 3]
```diff
@@ -0,0 +1,5 @@
+comparing
+= anObject
```
Comment (arminWel) [inline packages/CollabCodeEdit-Client.package/CCEClassDefinition.class/instance/^equals.st:2]: Should be specified

File: packages/CollabCodeEdit-Client.package/CCEClassDefinition.class/instance/equalsClass..st [hunk 4]
```diff
@@ -0,0 +1,9 @@
+comparing
+equalsClass: aClass
+	^  (self superclass = aClass superclass) &
+		(self subclass = aClass name asSymbol) &
```
Comment (arminWel) [inline packages/CollabCodeEdit-Client.package/CCEClassDefinition.class/instance/equalsClass..st:4]: Should be in []

File: packages/CollabCodeEdit-Tests.package/CCETestDummy.class/README.md [hunk 5]
```diff
@@ -1,4 +1,3 @@
-I'm a class, which is used for tests as a dummy class, which may be changed without risk.
+A CCETestDummy is xxxxxxxxx.
```
Comment (ADimeo) [inline packages/CollabCodeEdit-Tests.package/CCETestDummy.class/README.md]: Class comment is placeholder

File: packages/CollabCodeEdit-Client.package/CCEClassDefinition.class/instance/equalsClassDefinition..st [hunk 6]
```diff
@@ -0,0 +1,9 @@
+comparing
+equalsClassDefinition: aCCEClassDefinition
+	^ (self superclass name asSymbol = aCCEClassDefinition superclass name asSymbol) &
+		(self subclass = aCCEClassDefinition subclass) &
```
Comment (arminWel) [inline packages/CollabCodeEdit-Client.package/CCEClassDefinition.class/instance/equalsClassDefinition..st:4]: Should be in blocks.

File: packages/CollabCodeEdit-Tests.package/CCEPackageManagerTests.class/instance/setUp.st [hunk 7]
```diff
@@ -2,4 +2,4 @@ running
 setUp
 	super setUp.
 	client := CCEClientMock new.
-	packageManager listenForChangesInScope: self sessionScope for: client
```
Comment (ADimeo) [inline packages/CollabCodeEdit-Tests.package/CCEPackageManagerTests.class/instance/setUp.st]: Full stop at end of method

File: packages/CollabCodeEdit-Client.package/CCEClassDefinition.class/instance/superclass..st [hunk 8]
```diff
@@ -0,0 +1,3 @@
+accessing
+superclass: aClass
```
Comment (arminWel) [inline packages/CollabCodeEdit-Client.package/CCEClassDefinition.class/instance/superclass..st:2]: Do we want the superclass to be serialized as Class or just the Classname?

File: packages/CollabCodeEdit-Client.package/CCEEditBuilder.class/instance/buildClassDefinitionFrom..st [hunk 9]
```diff
@@ -0,0 +1,15 @@
...
+	
+	^ CCEClassDefinition 
+		superclass: class superclass
+		subclass: class name asSymbol 
+		instVarNames: class instanceVariablesString 
+		classVarNames: class classVariablesString 
+		poolDictionaries: class sharedPoolsString 
+		category: class category 
+		comment: class comment
+
```
Comment (arminWel) [inline packages/CollabCodeEdit-Client.package/CCEEditBuilder.class/instance/buildClassDefinitionFrom..st]: Trailing whitespace.

File: packages/CollabCodeEdit-Tests.package/CCEEditBuilderTest.class/instance/testBuildCCEMethodRecategorizedEdit.st [hunk 10]
```diff
@@ -0,0 +1,16 @@
+methodedit tests
+testBuildCCEMethodRecategorizedEdit
+	| event edit methodSelector |
+	
+	methodSelector := #testMethod.
+	event := (RecategorizedEvent method: (CCETestDummy compiledMethodAt: methodSelector) selector: methodSelector protocol: 'newCategory' class: CCETestDummy).
```
Comment (ADimeo) [inline packages/CollabCodeEdit-Tests.package/CCEEditBuilderTest.class/instance/testBuildCCEMethodRecategorizedEdit.st]: This line is to long

File: packages/CollabCodeEdit-Tests.package/CCEEditBuilderTest.class/instance/testBuildCCEMethodRemovedEdit.st [hunk 11]
```diff
@@ -0,0 +1,15 @@
+methodedit tests
+testBuildCCEMethodRemovedEdit
+	| event edit methodSelector |
+	
+	methodSelector := #testMethod.
+	event := RemovedEvent method: (CCETestDummy compiledMethodAt: methodSelector) selector: methodSelector class: CCETestDummy.
```
Comment (ADimeo) [inline packages/CollabCodeEdit-Tests.package/CCEEditBuilderTest.class/instance/testBuildCCEMethodRemovedEdit.st]: This line is to long

File: packages/CollabCodeEdit-Tests.package/CCEEditBuilderTest.class/instance/testBuildCCEMethodModifiedEdit.st [hunk 12]
```diff
@@ -0,0 +1,15 @@
...
+	
+	methodText := (CCETestDummy compiledMethodAt: #testMethod) getSource.
+	event := ModifiedEvent method: (CCETestDummy compiledMethodAt: #testMethod) class: CCETestDummy.
+	
+	edit := builder buildMethodEditFrom: event.
+	
+	self assert: edit editContent equals: methodText.
+	self assert: (PackageOrganizer default packageOfClass: CCETestDummy) equals: edit package.
+	self assert: (edit isKindOf: CCEMethodAddedOrModifiedEdit)
+
```
Comment (ADimeo) [inline packages/CollabCodeEdit-Tests.package/CCEEditBuilderTest.class/instance/testBuildCCEMethodModifiedEdit.st]: Trailing newlines

File: packages/CollabCodeEdit-Tests.package/CCEEditBuilderTest.class/instance/testBuildCCEMethodAddedEdit.st [hunk 13]
```diff
@@ -0,0 +1,15 @@
...
+	
+	methodText := (CCETestDummy compiledMethodAt: #testMethod) getSource.
+	event := AddedEvent method: (CCETestDummy compiledMethodAt: #testMethod) class: CCETestDummy.
+	
+	edit := builder buildMethodEditFrom: event.
+	
+	self assert: edit editContent equals: methodText.
+	self assert: (PackageOrganizer default packageOfClass: CCETestDummy) equals: edit package.
+	self assert: (edit isKindOf: CCEMethodAddedOrModifiedEdit)
+
```
Comment (ADimeo) [inline packages/CollabCodeEdit-Tests.package/CCEEditBuilderTest.class/instance/testBuildCCEMethodAddedEdit.st]: Trailing newlines

File: packages/CollabCodeEdit-Tests.package/CCEEditBuilderTest.class/instance/testBuildCCEClassRenamedEdit.st [hunk 14]
```diff
@@ -0,0 +1,16 @@
...
+	event := RenamedEvent class: CCETestDummy category: 'CollabCodeEdit-Tests' oldName: 'CCETestDummy1' newName: 'CCETestDummy'.
+	
+	edit := builder buildClassRenamedEditFrom: event.	
+
+	self assert: edit renamedClass equals: CCETestDummy.
+	self assert: edit oldName equals: 'CCETestDummy1'.
+	self assert: edit newName equals: 'CCETestDummy'.
+	self assert: (PackageOrganizer default packageOfClass: CCETestDummy) equals: edit package.
+	self assert: (edit isKindOf: CCEClassRenamedEdit)
+
```
Comment (ADimeo) [inline packages/CollabCodeEdit-Tests.package/CCEEditBuilderTest.class/instance/testBuildCCEClassRenamedEdit.st]: Trailing newlines

File: packages/CollabCodeEdit-Tests.package/CCEEditBuilderTest.class/instance/testBuildCCEClassRemovedEdit.st [hunk 15]
```diff
@@ -0,0 +1,13 @@
...
+testBuildCCEClassRemovedEdit
+	| event edit |
+	
+	event := RemovedEvent item: CCETestDummy kind: RemovedEvent classKind.
+	edit := builder buildClassRemovedEditFrom: event.	
+
+	self assert: edit editContent equals: CCETestDummy name.
+	self assert: (PackageOrganizer default packageOfClass: CCETestDummy) equals: edit package.
+	self assert: (edit isKindOf: CCEClassRemovedEdit)
+
```
Comment (ADimeo) [inline packages/CollabCodeEdit-Tests.package/CCEEditBuilderTest.class/instance/testBuildCCEClassRemovedEdit.st]: Trailing newlines

File: packages/CollabCodeEdit-Tests.package/CCEEditBuilderTest.class/instance/testBuildCCEClassRecategorizedEdit.st [hunk 16]
```diff
@@ -0,0 +1,14 @@
...
+	| event edit |
+	
+	event := RecategorizedEvent class: CCETestDummy category: CCETestDummy category oldCategory: 'CollabCodeEdit-Tests'.
+	
+	edit := builder buildClassModifiedEditFrom: event.
+	
+	self assert: (edit editContent equalsClass: CCETestDummy).
+	self assert: (PackageOrganizer default packageOfClass: CCETestDummy) equals: edit package.
+	self assert: (edit isKindOf: CCEClassModifiedEdit)
+
```
Comment (ADimeo) [inline packages/CollabCodeEdit-Tests.package/CCEEditBuilderTest.class/instance/testBuildCCEClassRecategorizedEdit.st]: Trailing newlines

File: packages/CollabCodeEdit-Tests.package/CCEEditBuilderTest.class/instance/testBuildCCEClassDefinitionModifiedEdit.st [hunk 17]
```diff
@@ -0,0 +1,14 @@
...
+	| event edit |
+	
+	event := ModifiedClassDefinitionEvent classDefinitionChangedFrom: CCETestDummy to: CCEModel.
+	
+	edit := builder buildClassModifiedEditFrom: event.
+	
+	self assert: (edit editContent equalsClass: CCEModel).
+	self assert: (PackageOrganizer default packageOfClass: CCEModel) equals: edit package.
+	self assert: (edit isKindOf: CCEClassModifiedEdit)
+
```
Comment (ADimeo) [inline packages/CollabCodeEdit-Tests.package/CCEEditBuilderTest.class/instance/testBuildCCEClassDefinitionModifiedEdit.st]: Trailing newlines

File: packages/CollabCodeEdit-Tests.package/CCEEditBuilderTest.class/instance/testBuildCCEClassCommentedEdit.st [hunk 18]
```diff
@@ -0,0 +1,14 @@
...
+	| event edit |
+	
+	event := CommentedEvent item: CCETestDummy kind: CommentedEvent classKind.
+	
+	edit := builder buildClassModifiedEditFrom: event.
+	
+	self assert: (edit editContent equalsClass: CCETestDummy).
+	self assert: (PackageOrganizer default packageOfClass: CCETestDummy) equals: edit package.
+	self assert: (edit isKindOf: CCEClassModifiedEdit)
+
```
Comment (ADimeo) [inline packages/CollabCodeEdit-Tests.package/CCEEditBuilderTest.class/instance/testBuildCCEClassCommentedEdit.st]: Trailing newlines

File: packages/CollabCodeEdit-Tests.package/CCEEditBuilderTest.class/instance/testBuildCCEClassAddedEdit.st [hunk 19]
```diff
@@ -0,0 +1,14 @@
...
+	| event edit |
+	
+	event := AddedEvent item: CCETestDummy kind: AddedEvent classKind.
+	
+	edit := builder buildClassModifiedEditFrom: event.
+	
+	self assert: (edit editContent equalsClass: CCETestDummy).
+	self assert: (PackageOrganizer default packageOfClass: CCETestDummy) equals: edit package.
+	self assert: (edit isKindOf: CCEClassModifiedEdit)
+
```
Comment (ADimeo) [inline packages/CollabCodeEdit-Tests.package/CCEEditBuilderTest.class/instance/testBuildCCEClassAddedEdit.st]: Trailing newlines

File: packages/CollabCodeEdit-Tests.package/CCEClassEditTest.class/instance/testClassRemoved.st [hunk 20]
```diff
@@ -0,0 +1,15 @@
...
+	self addNewClassFromTestClassDefinition.
+	newlyCreatedClass := Smalltalk classNamed: testClassName.
+	
+	
+	(CCEClassRemovedEdit newWith: newlyCreatedClass name) apply.
+	
+	self assert: newlyCreatedClass isObsolete.
+	
+	(newlyCreatedClass isObsolete) ifFalse: [ newlyCreatedClass removeFromSystem ]
+
```
Comment (ADimeo) [inline packages/CollabCodeEdit-Tests.package/CCEClassEditTest.class/instance/testClassRemoved.st]: Trailiing newline

File: packages/CollabCodeEdit-Tests.package/CCEClassEditTest.class/instance/testClassRemoved.st [hunk 21]
```diff
@@ -0,0 +1,15 @@
+tests
+testClassRemoved
+
```
Comment (ADimeo) [inline packages/CollabCodeEdit-Tests.package/CCEClassEditTest.class/instance/testClassRemoved.st]: Newline where there should be none

File: packages/CollabCodeEdit-Tests.package/CCEClassEditTest.class/instance/testClassModified.st [hunk 22]
```diff
@@ -0,0 +1,22 @@
...
+	
+	newlyDefinedClass := Smalltalk classNamed: testClassName.
+	self deny: (newlyDefinedClass isObsolete).
+	
+	self assert: newlyDefinedClass superclass equals: testClassDefinition superclass.
+	self assert: newlyDefinedClass instanceVariablesString equals: testClassDefinition instVarNames.
+	self assert: newlyDefinedClass classVariablesString equals: testClassDefinition classVarNames.
+	self assert: newlyDefinedClass sharedPoolsString equals: testClassDefinition poolDictionaries.
+	self assert: newlyDefinedClass category equals: testClassDefinition category.
+	self assert: newlyDefinedClass comment equals: testClassDefinition comment.
```
Comment (ADimeo) [inline packages/CollabCodeEdit-Tests.package/CCEClassEditTest.class/instance/testClassModified.st]: Trailing full stop

File: packages/CollabCodeEdit-Client.package/CCEPackageManager.class/instance/classModified..st [hunk 23]
```diff
@@ -0,0 +1,3 @@
+capturing classes
+classModified: anEvent
+	self sendEditIfNotToBeIgnored: (CCEEditBuilder new buildClassModifiedEditFrom: anEvent)
```
Comment (arminWel) [inline packages/CollabCodeEdit-Client.package/CCEPackageManager.class/instance/classModified..st:3]: All Methods of these kind: Shouldnt the builder decide what edit to build?

File: packages/CollabCodeEdit-Tests.package/CCEClassEditTest.class/instance/testClassModified.st [hunk 24]
```diff
@@ -0,0 +1,22 @@
...
+
```
Comment (ADimeo) [inline packages/CollabCodeEdit-Tests.package/CCEClassEditTest.class/instance/testClassModified.st]: Trailing newlines

File: packages/CollabCodeEdit-Tests.package/CCEClassEditTest.class/instance/addNewClassFromTestClassDefinition.st [hunk 25]
```diff
@@ -0,0 +1,5 @@
+helper
+addNewClassFromTestClassDefinition
+	CCEClassModifiedEdit new 
+		editContent: testClassDefinition;
+		apply.
```
Comment (ADimeo) [inline packages/CollabCodeEdit-Tests.package/CCEClassEditTest.class/instance/addNewClassFromTestClassDefinition.st]: Trailing fullstop

File: packages/CollabCodeEdit-Core.package/CCEEdit.class/instance/securelyApplyIn..st [hunk 26]
```diff
@@ -1,8 +1,8 @@
 applying
 securelyApplyIn: aCCESession
 	(self mayBeAppliedIn: aCCESession)
-		ifTrue: [ 
-			aCCESession packageManager ignoreEdit: self.
+		ifTrue: [ 			
+			aCCESession packageManager ignoreEdit: self.
```
Comment (ADimeo) [inline packages/CollabCodeEdit-Core.package/CCEEdit.class/instance/securelyApplyIn..st]: Trailing spaces

File: packages/CollabCodeEdit-Client.package/CCEPackageManager.class/instance/listenForClassChanges.st [hunk 27]
```diff
@@ -0,0 +1,15 @@
+capturing classes
+listenForClassChanges
+	SystemChangeNotifier uniqueInstance 
+		notify: self ofSystemChangesOfItem: #class change: #Added
```
Comment (arminWel) [inline packages/CollabCodeEdit-Client.package/CCEPackageManager.class/instance/listenForClassChanges.st:4]: This should be handled solely be the editbuilder

File: packages/CollabCodeEdit-Core.package/CCEClassRemovedEdit.class/instance/apply.st [hunk 28]
```diff
@@ -0,0 +1,3 @@
+as yet unclassified
```
Comment (ADimeo) [inline packages/CollabCodeEdit-Core.package/CCEClassRemovedEdit.class/instance/apply.st]: Method unclassified

File: packages/CollabCodeEdit-Core.package/CCEClassModifiedEdit.class/instance/applySmalltalkClassDefinitionFrom..st [hunk 29]
```diff
@@ -0,0 +1,10 @@
+applying
+applySmalltalkClassDefinitionFrom: aCCEClassDefinition
+	ClassBuilder new 
+		superclass: aCCEClassDefinition superclass
+		subclass: aCCEClassDefinition subclass
+		instanceVariableNames: aCCEClassDefinition instVarNames
+		classVariableNames: aCCEClassDefinition classVarNames
+		poolDictionaries: aCCEClassDefinition poolDictionaries
+		category: aCCEClassDefinition category
+
```
Comment (ADimeo) [inline packages/CollabCodeEdit-Core.package/CCEClassModifiedEdit.class/instance/applySmalltalkClassDefinitionFrom..st]: Trailing newline

File: packages/CollabCodeEdit-Client.package/CCESession.class/instance/mayBeApplied..st [hunk 30]
```diff
@@ -1,3 +1,3 @@
 as yet unclassified
```
Comment (arminWel) [inline packages/CollabCodeEdit-Client.package/CCESession.class/instance/mayBeApplied..st]: .

File: packages/CollabCodeEdit-Core.package/CCEClassModifiedEdit.class/instance/apply.st [hunk 31]
```diff
@@ -0,0 +1,8 @@
+applying
+apply
+	| classDefinition |
```
Comment (ADimeo) [inline packages/CollabCodeEdit-Core.package/CCEClassModifiedEdit.class/instance/apply.st]: Unnecessary local variable

File: packages/CollabCodeEdit-Core.package/CCEClassModifiedEdit.class/instance/apply.st [hunk 32]
```diff
@@ -0,0 +1,8 @@
+applying
+apply
+	| classDefinition | 
+	
+	classDefinition := self editContent.
+	
+	self applySmalltalkClassDefinitionFrom: classDefinition;
+		applyCommentFrom: classDefinition.
```
Comment (ADimeo) [inline packages/CollabCodeEdit-Core.package/CCEClassModifiedEdit.class/instance/apply.st]: Trailing fullstop

File: packages/CollabCodeEdit-Core.package/CCEClassModifiedEdit.class/instance/applyCommentFrom..st [hunk 33]
```diff
@@ -0,0 +1,3 @@
+applying
+applyCommentFrom: aCCEClassDefinition
+	(Smalltalk classNamed: aCCEClassDefinition subclass asString) comment: aCCEClassDefinition comment
```
Comment (arminWel) [inline packages/CollabCodeEdit-Core.package/CCEClassModifiedEdit.class/instance/applyCommentFrom..st]: We does the asString do here?

File: packages/CollabCodeEdit-Core.package/CCEClassRemovedEdit.class/instance/apply.st (hunk 28 above)
Comment (arminWel) [inline packages/CollabCodeEdit-Core.package/CCEClassRemovedEdit.class/instance/apply.st]: .

File: packages/CollabCodeEdit-Core.package/CCEClassRenamedEdit.class/instance/apply.st [hunk 34]
```diff
@@ -0,0 +1,3 @@
+as yet unclassified
```
Comment (arminWel) [inline packages/CollabCodeEdit-Core.package/CCEClassRenamedEdit.class/instance/apply.st]: .

File: packages/CollabCodeEdit-Core.package/CCEClassRenamedEdit.class/instance/apply.st [hunk 35]
```diff
@@ -0,0 +1,3 @@
...
+apply
+	(Smalltalk classNamed: (self oldName)) rename: self newName
```
Comment (arminWel) [inline packages/CollabCodeEdit-Core.package/CCEClassRenamedEdit.class/instance/apply.st]: Unnecessary brackets.

File: packages/CollabCodeEdit-Client.package/CCEClassDefinition.class/instance/^equals.st (hunk 3 above)
Comment (ADimeo) [inline packages/CollabCodeEdit-Client.package/CCEClassDefinition.class/instance/^equals.st:2]: Should this be this generic (anObject)?

File: packages/CollabCodeEdit-Client.package/CCEClassDefinition.class/class/superclass.subclass.instVarNames.classVarNames.poolDictionaries.category.comment..st (hunk 1 above)
Comment (ADimeo) [inline packages/CollabCodeEdit-Client.package/CCEClassDefinition.class/class/superclass.subclass.instVarNames.classVarNames.poolDictionaries.category.comment..st]: As yet unclassified

**Review by ADimeo — CHANGES_REQUESTED** [review]

File: packages/CollabCodeEdit-Core.package/CCEEdit.class/instance/package..st [hunk 36]
```diff
@@ -0,0 +1,3 @@
+accessing
+package: aString
+	package := aString
```
Comment (arminWel) [inline packages/CollabCodeEdit-Core.package/CCEEdit.class/instance/package..st]: Is this really a String or rather a the package?

File: packages/CollabCodeEdit-Tests.package/CCEClassEditTest.class/instance/addNewClassFromTestClassDefinition.st (hunk 25 above)
Comment (arminWel) [inline packages/CollabCodeEdit-Tests.package/CCEClassEditTest.class/instance/addNewClassFromTestClassDefinition.st]: Trailing dot.

File: packages/CollabCodeEdit-Tests.package/CCEClassEditTest.class/instance/setUp.st [hunk 37]
```diff
@@ -0,0 +1,14 @@
+running
+setUp
+	super setUp.
+	
+	testClassName := 'TestCaseGeneratedClass'.
+	testClassSymbol := testClassName asSymbol.
+	testClassDefinition := CCEClassDefinition
```
Comment (arminWel) [inline packages/CollabCodeEdit-Tests.package/CCEClassEditTest.class/instance/setUp.st]: Long method. Maybe a `createTestClassDefinition` would help.

File: packages/CollabCodeEdit-Tests.package/CCEClassEditTest.class/instance/testClassModified.st (hunk 22 above)
Comment (arminWel) [inline packages/CollabCodeEdit-Tests.package/CCEClassEditTest.class/instance/testClassModified.st]: Trailing dot.

File: packages/CollabCodeEdit-Tests.package/CCEClassEditTest.class/instance/testClassRemoved.st [hunk 38]
```diff
@@ -0,0 +1,15 @@
...
+	
+	self addNewClassFromTestClassDefinition.
+	newlyCreatedClass := Smalltalk classNamed: testClassName.
+	
+	
+	(CCEClassRemovedEdit newWith: newlyCreatedClass name) apply.
+	
+	self assert: newlyCreatedClass isObsolete.
+	
+	(newlyCreatedClass isObsolete) ifFalse: [ newlyCreatedClass removeFromSystem ]
```
Comment (arminWel) [inline packages/CollabCodeEdit-Tests.package/CCEClassEditTest.class/instance/testClassRemoved.st]: This line will never be called. Use ensure:

File: packages/CollabCodeEdit-Tests.package/CCEClassEditTest.class/instance/testClassRenamed.st [hunk 39]
```diff
@@ -0,0 +1,17 @@
...
+	CCEClassRenamedEdit new
+		renamedClass: (Smalltalk classNamed: testClassName);
+		newName: newName;
+		oldName: testClassName;
+		apply.
+		
+	self deny: ((Smalltalk classNamed: newName) isNil).
+	self assert: ((Smalltalk classNamed: testClassName) isNil).
+	
+	(Smalltalk classNamed: newName) ifNotNil: [ :class | class removeFromSystem ]
```
Comment (arminWel) [inline packages/CollabCodeEdit-Tests.package/CCEClassEditTest.class/instance/testClassRenamed.st]: Use ensure:

File: packages/CollabCodeEdit-Tests.package/CCEEditBuilderTest.class/instance/testBuildCCEClassCommentedEdit.st [hunk 40]
```diff
@@ -0,0 +1,14 @@
+classedit tests
+testBuildCCEClassCommentedEdit
+	| event edit |
+	
+	event := CommentedEvent item: CCETestDummy kind: CommentedEvent classKind.
+	
+	edit := builder buildClassModifiedEditFrom: event.
+	
+	self assert: (edit editContent equalsClass: CCETestDummy).
```
Comment (arminWel) [inline packages/CollabCodeEdit-Tests.package/CCEEditBuilderTest.class/instance/testBuildCCEClassCommentedEdit.st:9]: These asserts are quite common in these tests. Create a custom assert for it.

File: packages/CollabCodeEdit-Tests.package/CCEEditBuilderTest.class/instance/testBuildCCEMethodRemovedEdit.st (hunk 11 above)
Comment (arminWel) [inline packages/CollabCodeEdit-Tests.package/CCEEditBuilderTest.class/instance/testBuildCCEMethodRemovedEdit.st]: Long line

File: packages/CollabCodeEdit-Tests.package/CCEPackageManagerTests.class/instance/setUp.st [hunk 41]
```diff
@@ -2,4 +2,4 @@ running
...
\ No newline at end of file
+	packageManager listenForChangesInScope: self sessionScope for: client.
```
Comment (arminWel) [inline packages/CollabCodeEdit-Tests.package/CCEPackageManagerTests.class/instance/setUp.st]: What the Change?

File: packages/CollabCodeEdit-Tests.package/CCETestDummy.class/instance/testMethodScope.st [hunk 42]
```diff
@@ -1,3 +1,2 @@
 as yet unclassified
```
Comment (arminWel) [inline packages/CollabCodeEdit-Tests.package/CCETestDummy.class/instance/testMethodScope.st:1]: .

**Review by ADimeo — APPROVED** [review]

**Pull request merged**